misk-aws/src/main/kotlin/misk/jobqueue/sqs/SqsJobConsumer.kt

```diff
@@ -0,0 +1,117 @@
+package misk.jobqueue.sqs
+
+import com.amazonaws.services.sqs.AmazonSQS
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest
+import com.google.common.util.concurrent.AbstractIdleService
+import io.opentracing.Tracer
+import io.opentracing.tag.StringTag
+import io.opentracing.tag.Tags
+import misk.jobqueue.JobConsumer
+import misk.jobqueue.JobHandler
+import misk.jobqueue.QueueName
+import misk.logging.getLogger
+import misk.time.timed
+import misk.tracing.traceWithSpan
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.atomic.AtomicBoolean
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.concurrent.thread
+
+@Singleton
+internal class SqsJobConsumer @Inject internal constructor(
+  private val config: AwsSqsJobQueueConfig,
+  private val sqs: AmazonSQS,
+  private val queueUrls: QueueUrlMapping,
+  @ForSqsConsumer private val dispatchThreadPool: ExecutorService,
+  private val tracer: Tracer,
+  private val metrics: SqsMetrics
+) : AbstractIdleService(), JobConsumer {
```

Comment:
any opinion if we should use the other types of guava Services that help with thread management?

https://github.com/google/guava/wiki/ServiceExplained